kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/engine/LoggingMetricsReporter.java

```diff
@@ -47,6 +44,11 @@ public void report(MetricsReport report) {
         logger.info(
             "TransactionReport = {}",
             MetricsReportSerializers.serializeTransactionReport((TransactionReport) report));
+      } else if (report instanceof PostCommitHookMetricsReport) {
```

Comment:
Why don't we push all the `instanceof`ing down into a `serializeMetricsReport` method? Probably cc @allisonport-db